typhos/benchmark/device.py

```diff
@@ -0,0 +1,50 @@
+"""
+A generator of test devices.
+
+These are meant to be used for targetted benchmarking and typically are
+"extreme" in both size and in composition in order to make specific
+loading issues more obvious.
+
+Note that this currently only supports simple, flat devices.
+In the future it can be expanded to have complicated nesting devices,
+Kind information, different data types to test different widget types, etc.
+"""
+from ophyd.device import Device, Component as Cpt
+from ophyd.signal import Signal
+
+
+def make_test_device_class(name='TestClass', signal_class=Signal,
+                           include_prefix=False, num_signals=10):
+    """
+    Creates a test Device class.
+
+    Parameters
+    ----------
+    name: str, optional
+        The name of the class.
+        Defaults to 'TestClass'.
+
+    signal_class: type, optional
+        Picks which type of signal to use in the Device.
+        Defaults to ophyd.signal.Signal.
```

Comment:
```suggestion
        Defaults to :class:`ophyd.Signal`.
```

Reply:
What effect does this have on the generated docs?

Reply:
I think with the new configuration Tyler's working on, we'll be able to cross-link into ophyd documentation.